### ISSUE TITLE
Unable to reproduce zeroshot classification results

### ISSUE THREAD

**Issue by cyrusvahidi** [LAION-AI/CLAP#133]
I have attempted to reproduce the zeroshot classification results for ESC-50 outlined in the publication [Large-scale contrastive language-audio pretraining with feature fusion and keyword-to-caption augmentation](https://ieeexplore.ieee.org/abstract/document/10095969/?casa_token=vaS8vCpiQVMAAAAA:9afRPbF-y-vDgJwVHZl1mCorHwxuzHHUXGIXJTQifkT_XxBE1MJ2GsMlyDPlVUgy8SHlMrudjA).

Zeroshot classification accuracy (top-1) is reported at `91.0%` for the best model that uses K2C augmentation. I assume that this is the default model checkpoint that's shipped with the software (`630k-audioset-best.pt`).

Somehow I am only able to report `60.2%` top-1 accuracy for the ESC-50 dataset, which I downloaded from the provided [google drive](https://drive.google.com/drive/folders/1scyH43eQAcrBz-5fAw44C6RNBhC3ejvX) folder. I've found it quite difficult to run the provided evaluation and data preprocessing scripts, so I wrote my own.

I use the set of 50 unique captions in the test dataset, which are found in the `text` attr of each json file, e.g. `"The sound of the crow"`. 

Here's the loader for `ESC-50`:

```python
class ESC50Dataset(Dataset):
    def __init__(
        self,
        path_to_esc50="./data/ESC50",
        split="test",
        audio_len = 480000,
    ):
        super().__init__()
        self.data_path = Path(path_to_esc50)
        self.audio_len = audio_len
       
        self.audio_files = sorted(glob.glob(str(self.data_path / split / "*.flac")))
        self.meta_files = sorted(glob.glob(str(self.data_path / split / "*.json")))
        assert len(self.audio_files) == len(self.meta_files), "Number of audio files and meta files must match"
        assert [osp.splitext(osp.basename(x))[0] for x in self.audio_files] == [osp.splitext(osp.basename(x))[0] for x in self.meta_files], "Audio files and meta files must have the same names"

        self.tags = []
        self.texts = []
        for f in self.meta_files:
            with open(f, 'r') as json_file:
                data = json.load(json_file)
                self.tags.append(data["tag"][0])
                self.texts.append(data["text"][0])

    def __getitem__(self, idx):
        x, _ = load_audio_torch(self.audio_files[idx], target_sr=48000, mono=True)
        x = random_slice(x, self.audio_len)
        return x, self.texts[idx]

    def __len__(self):
        return len(self.audio_files)
```

And the zeroshot retrieval script:

```python
import os

import torch
import laion_clap

from data.loaders import ESC50Dataset

ckpt_path = "CLAP_checkpoints/laion_clap/"
model_params = {"ckpt": "630k-audioset-best.pt", "amodel": "HTSAT-tiny"}
model = laion_clap.CLAP_Module(enable_fusion=False, amodel=model_params["amodel"])
model.load_ckpt(os.path.join(ckpt_path, model_params["ckpt"]))

dataset = ESC50Dataset()
texts = list(set(dataset.texts)) # get the unique texts, e.g "The sound of the crow"

# get the text embeddings for each tag
z_text = torch.cat([torch.tensor(model.get_text_embedding([t, t])[0:1]) for t in texts])

z_audio = []
text_idxs = []
for item in dataset:
    x, text = item
    idx = texts.index(text) # get the index of this example's text
    text_idxs.append(idx)
    z_audio.append(torch.tensor(model.get_audio_embedding_from_data(x.numpy()))) # get its CLAP audio embedding
z_audio = torch.cat(z_audio)
sim = model.model.logit_scale_a.cpu() * z_audio @ z_text.T # compute pairwise dot products

# top-1 accuracy
acc = float(torch.sum(torch.argmax(sim, dim=1) == torch.tensor(text_idxs)) / len(sim))
print(f"Accuracy: {acc}")
```
```bash
Accuracy: 0.6025000214576721 
```

Hopefully I am missing something significant?
